src/HolographicCamera.Unity/Assets/HolographicCamera/Scripts/HeadsetRequestHandler.cs

```diff
@@ -0,0 +1,193 @@
+        /// </summary>
+        [Tooltip("Used to send and receive network messages.")]
+        [SerializeField]
+        protected HolographicCameraBroadcaster holographicCameraBroadcaster;
+
+        [SerializeField]
+        private TCPConnectionManager connectionManager = null;
+
+        /// <summary>
+        /// Used to obtain marker and headset data.
+        /// </summary>
+        [Tooltip("Used to obtain marker and headset data.")]
+        [SerializeField]
+        protected HeadsetCalibration headsetCalibration;
+
+        /// <summary>
+        /// Used to display the last request timestamp.
+        /// </summary>
+        [Tooltip("Used to display the last request timestamp.")]
+        [SerializeField]
+        protected Text lastRequestTimestampText;
+
+        private bool initialized = false;
+        private string editorAddress = string.Empty;
+        private bool updateData = false;
+        private HeadsetCalibrationDataRequest request = null;
+
+        private void OnEnable()
+        {
+            holographicCameraBroadcaster.Disconnected += OnDisconnect;
+            holographicCameraBroadcaster.RegisterCommandHandler(HeadsetCalibration.RequestCalibrationDataCommandHeader, CalibrationDataRequested);
+            holographicCameraBroadcaster.RegisterCommandHandler(HeadsetCalibration.UploadCalibrationCommandHeader, UploadCalibrationDataAsync);
+            headsetCalibration.Updated += OnHeadsetCalibrationUpdated;
+        }
+
+        private void EnableChildren()
+        {
+            if (!initialized)
+            {
+                foreach (Transform child in transform)
+                {
+                    child.gameObject.SetActive(true);
+                }
+
+                initialized = true;
+            }
+        }
+
+        private void DisableChildren()
+        {
+            if (initialized)
+            {
+                foreach (Transform child in transform)
+                {
+                    child.gameObject.SetActive(false);
+                }
+
+                initialized = false;
+            }
+        }
+
+        private void OnDisconnect(SocketEndpoint endpoint)
+        {
+            if(endpoint.Address == editorAddress)
+            {
+                DisableChildren();
+            }
+        }
+
+        private void CalibrationDataRequested(SocketEndpoint endpoint, string command, BinaryReader reader, int remainingDataSize)
+        {
+            editorAddress = endpoint.Address;
+            EnableChildren();
+
+            if (HeadsetCalibrationDataRequest.TryDeserialize(reader, out request))
+            {
+                Debug.Log("Headset calibration data requested");
+                updateData = true;
+            }
+            else
+            {
+                Debug.LogWarning("Received network payload that wasn't a headset calibration data request");
+                request = null;
+            }
+        }
+
+#if WINDOWS_UWP

```

Comment:
>WINDOWS_UWP [](start = 4, length = 11)

I know in the past I've had problems with WINDOWS_UWP being defined at weird parts of the Unity build process, where UNITY_WSA is always defined at the correct times. I'd double check that every build configuration (.NET Scripting + IL2CPP) works with this define, or switch it to UNITY_WSA

Reply:
Tested for both .net and il2cpp